addons/mail/models/discuss/discuss_channel.py

```diff
@@ -438,8 +438,8 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                 member._bus_send("discuss.channel/joined", payload)
                 if post_joined_message:
                     notification = (
-                        _("joined the channel")
-                        if member.is_self
+                        _("joined the thread") if member.is_self and self.parent_channel_id
+                        else _("joined the channel") if member.is_self
                         else _("invited %s to the channel", member._get_html_link(for_persona=True))
```

Comment:
Would take the opportunity to reformat notification as a full sentence, `%(person)s joined the {channel/thread}`, `%(person1) invited %(person2) to the channel`, etc. This would make format of string good in all languages.

This reduces the amount of occurrences of bad JS concatenation of author name and notification text in JS by 1. Eventually the code in JS should be removed, I think there'll be only "started a call meeting" and we can remove the crappy JS code :)